File: fixtures/adapter-java/basic/src/main/java/com/example/Elsewhere.java
package com.example;

// UserService is declared in another file of the same package, so Java needs
// no import for it — and neither does the resolver. Both edges below are
// cross-file, which is exactly what Tier 2 could never record.
public class Elsewhere {
    public void run() {
        UserService service = new UserService();
        service.handle();
    }
}

File: fixtures/adapter-csharp/basic/Elsewhere.cs
// UserService is declared in another file of the same namespace, so C# needs
// no using for it — and neither does the resolver. `var` here names its type
// in the initialiser, which is read rather than inferred.
namespace Example
{
    public class Elsewhere
    {
        public void Run()
        {
            var service = new UserService();
            service.Handle();
        }
    }
}
